src/utils/adminPluginInstaller.ts

```
@@ -0,0 +1,430 @@
+    if (!structure.hasAdminFolder && !structure.hasApiFolder) {
+      return {
+        success: false,
+        pluginId: '',
+        manifest: {} as AdminPluginManifest,
+        installedComponents: [],
+        error:
+          "Zip file must contain either 'admin' or 'api' folder with valid plugin structure",
+      };
+    }
+
+    const pluginId = structure.pluginId!;
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)